src/simulation/hdf_simulation_store.jl

```diff
@@ -379,14 +379,180 @@ function read_result(
     return data, columns
 end
 
+function write_model_dual_results!(
```

Comment:
This file used to be generic with regard to dual/parameter/variable/aux_variable. Why did you have to move it here? It means that similar logic would have to be duplicated in all subtypes of SimulationStore.

Reply:
I moved it because the logic is now different between the store types, in order to implement the InMemoryModel Store I had to specify the store types and dispatch on those so I thought it best to move this implementation here.

Reply:
To be done in the store refactoring